pkg/libursa/ursa/value_builder.go

```diff
@@ -0,0 +1,91 @@
+package ursa
+
+/*
+   #cgo LDFLAGS: -lursa
+   #include "ursa_cl.h"
+   #include <stdlib.h>
+*/
+import "C"
+import (
+	"unsafe"
+)
+
+// NewValuesBuilder creates and returns credentials values entity builder
+func NewValueBuilder() (unsafe.Pointer, error) {
+	var builder unsafe.Pointer
+
+	result := C.ursa_cl_credential_values_builder_new(&builder)
+	if result.code != 0 {
+		return nil, ursaError(C.GoString(result.message))
+	}
+
+	return builder, nil
+}
+
+// AddDecHidden adds new hidden attribute dec_value to credential values map
+func AddDecHidden(builder unsafe.Pointer, attr, decValue string) error {
+	cattr := C.CString(attr)
+	cval := C.CString(decValue)
+
+	result := C.ursa_cl_credential_values_builder_add_dec_hidden(builder, cattr, cval)
+	if result.code != 0 {
+		return ursaError(C.GoString(result.message))
+	}
+	C.free(unsafe.Pointer(cattr))
+	C.free(unsafe.Pointer(cval))
+
```

Comment:
These two frees should be defer'ed and moved above the C call so they are still called if the method fails